virtcontainers/api.go

```diff
@@ -74,6 +74,13 @@ func createSandboxFromConfig(ctx context.Context, sandboxConfig SandboxConfig, f
 		return nil, err
 	}
 
+	// cleanup sandbox resources in case of any failure
+	defer func() {
+		if err != nil {
+			s.Delete()
```

Comment:
I'm afraid is it reasonable to do sandbox.Delete() if it failed in fetching an existing sandbox instead of creating a new one.

Reply:
This is the create sandbox API.